src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -723,6 +723,7 @@ protected async Task SingleTopologyUpdateAsync(string[] initialUrls, Guid? appli
                         await ExecuteAsync(serverNode, null, context, command, shouldRetry: false, sessionInfo: null, token: CancellationToken.None)
                             .ConfigureAwait(false);
                         serverNode.ClusterTag = command.Result.NodeTag;
+                        serverNode.ServerRole = ServerNode.Role.Member;
```

Comment:
I think we need to take whatever the server sends us...

Reply:
It doesn't send the server role for this url. It does send the entire cluster topology but then we'll have to look up the url inside the topology and it might look different (private/public, diff format etc)

Reply:
so why not to change the EP to send us the role as well?